src/Sarif.UnitTests/Core/StackTests.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.
+// Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Linq;
+using System.IO;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    [TestClass]
+    public class StackTests
+    {
+        [TestMethod]
+        public void Stack_CreateFromStackTrace()
+        {
+            var dotNetStack = new StackTrace();
+            Stack stack = Stack.Create(dotNetStack);
+
+            // The .NET StackTrace.ToString() override must preserve a trailing NewLine
+            // for compatibility reasons. We do not retain this behavior in ToString()
+            // but provide a facility for adding the trailing NewLine
+            Assert.AreEqual(dotNetStack.ToString(), stack.ToString(StackFormat.TrailingNewLine));
```

Comment:
I personally like Assert.AreEqual(x, y) better than x.Should().Be(y).  I guess it's fine for unit tests, but I've seen a lot of crazy code backing fluent API and they get my head spinning around "what does Should()" do exactly?".  This line of code is roughly if (!Object.Equals(x, y)) throw new AssertException(); I personally don't see the value in making that into an English sentence. I can read code. ;)



---
In reply to: [60628907](https://github.com/Microsoft/sarif-sdk/pull/205#discussion_r60628907) [](ancestors = 60628907)